src/item.cpp

```diff
@@ -6921,6 +6921,16 @@ bool item::is_reloadable_helper( const itype_id &ammo, bool now ) const
         return true;
     }
 
+    if( is_watertight_container() && !contents.empty() &&
+        ( contents.legacy_front().typeId() == ammo ) ) {
```

Comment:
I wouldn't add new code that calls a function which says "legacy" in its name. I think you can use `item_contents::num_item_stacks` and `item_contents::only_item` instead.

Reply:
Good point. I didn't even think about that.